test/new-e2e/tests/npm/ec2_1host_test.go

```diff
@@ -0,0 +1,173 @@
+	// Other VM image can be used, our kitchen CI images test/kitchen/platforms.json
+	// ec2params.WithImageName("ami-a4dc46db", os.AMD64Arch, ec2os.AmazonLinuxOS) // ubuntu-16-04-4.4
+	e2e.Run(t, s, e2e.FakeIntakeStackDef(e2e.WithAgentParams(agentparams.WithSystemProbeConfig(NPMsystemProbeConfig))), e2eParams...)
+}
+
+// TestFakeIntakeNPM Validate the agent can communicate with the (fake) backend and send connections every 30 seconds
+//   - looking for 1 host to send CollectorConnections payload to the fakeintake
+//   - looking for 3 payloads and check if the last 2 have a span of 30s +/- 500ms
+func (v *ec2VMSuite) TestFakeIntakeNPM() {
+	t := v.T()
+
+	// default is to reset the current state of the fakeintake aggregators
+	if !v.DevMode {
+		v.Env().Fakeintake.FlushServerAndResetAggregators()
+	}
+
+	targetHostnameNetID := ""
+	// looking for 1 host to send CollectorConnections payload to the fakeintake
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		// generate a connection
+		v.Env().VM.Execute("curl http://www.datadoghq.com")
+
+		hostnameNetID, err := v.Env().Fakeintake.GetConnectionsNames()
+		assert.NoError(c, err, "GetConnectionsNames() errors")
+		if !assert.NotZero(c, len(hostnameNetID), "no connections yet") {
+			return
+		}
+		targetHostnameNetID = hostnameNetID[0]
+
+		t.Logf("hostname+networkID %v seen connections", hostnameNetID)
+	}, 60*time.Second, time.Second, "no connections received")
+
+	// looking for 3 payloads and check if the last 2 have a span of 30s +/- 500ms
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		cnx, err := v.Env().Fakeintake.GetConnections()
+		assert.NoError(t, err)
+
+		if !assert.Greater(c, len(cnx.GetPayloadsByName(targetHostnameNetID)), 2, "not enough payloads") {
+			return
+		}
+		var payloadsTimestamps []time.Time
+		for _, cc := range cnx.GetPayloadsByName(targetHostnameNetID) {
+			payloadsTimestamps = append(payloadsTimestamps, cc.GetCollectedTime())
+		}
+		dt := payloadsTimestamps[2].Sub(payloadsTimestamps[1]).Seconds()
+		t.Logf("hostname+networkID %v diff time %f seconds", targetHostnameNetID, dt)
+
+		// we want the test fail now, not retrying on the next payloads
+		require.Greater(t, 0.5, math.Abs(dt-30), "delta between collection is higher that 500ms")
+	}, 90*time.Second, time.Second, "not enough connections received")
+}
+
+// TestFakeIntakeNPM_TCP_UDP_DNS validate we received tcp, udp, and DNS connections
+// with some basic checks, like IPs/Ports present, DNS query has been captured, ...
+func (v *ec2VMSuite) TestFakeIntakeNPM_TCP_UDP_DNS() {
+	t := v.T()
+
+	// default is to reset the current state of the fakeintake aggregators
+	if !v.DevMode {
+		v.Env().Fakeintake.FlushServerAndResetAggregators()
+	}
+
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		// generate connections
+		v.Env().VM.Execute("curl http://www.datadoghq.com")
+		v.Env().VM.Execute("dig @8.8.8.8 www.google.ch")
+
+		cnx, err := v.Env().Fakeintake.GetConnections()
```

Comment:
Do we need to wait for the connections from 109/110 to get sent to the backend before asking FakeIntake for connections?

Reply:
Ideally we should have a first eventually loop that waits for `system-probe` to be ready, then trigger the action once, and loop again waiting for the payload to reach fakeintake.

Reply:
https://github.com/DataDog/datadog-agent/pull/19449/files#r1338832539

Reply:
yes, but we cannot know how long to wait, or we will need to switch the test in the future like this
 * wait system-probe to really start (/debug/net_maps)
 * wait cmd curl to run
 * wait 30 + 1 seconds
 * fakeIntake.GetConnections()